JUPE7H/Logic/Array2DExtensions.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace JUPE7H.Logic;
+
+internal static class Array2DExtensions{
+    public static void Fill<T>(this T[,] array, T value) where T: struct{
+        for(int x = 0; x < array.GetLength(0); x++){
+            for(int y = 0; y < array.GetLength(1); y++){
+                array[x, y] = value;
+            }
+        }
+    }
+    
+    public static void CopyTo<T>(this T[,] source, T[,] destination) where T: struct{
+        if(source.GetLength(0) != destination.GetLength(0) || source.GetLength(1) != destination.GetLength(1)){
+            throw new IndexOutOfRangeException("Source and destination sizes must match");
```

Comment:
`IndexOutOfRangeException`-t nem dobunk user kódból. Ez az exception tömb túl és alulindexelésnél dobódik amúgy, szóval nem picit félrevezető, de ez mondtam is órán. Esetleg nem voltál bent?